packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -351,21 +395,165 @@ Future<List<Plugin>> findPlugins(FlutterProject project, { bool throwOnError = t
     project.directory.path,
     '.packages',
   );
+  final String pubspec = globals.fs.path.join(
```

Comment:
This sort of parsing should be a part of the FlutterProject or FlutterManifest classes, since those already have the mechanisms for validating the yaml.

Reply:
It validates the Flutter plugins pubspec in conjunction with the app's pubspec.

Reply:
I'm not quite sure what you mean.

Reply:
Please don't re-parse the pubspec here. Lookup this information from the FlutterProject and add it if it is missing.